src/App.tsx

```diff
@@ -1,14 +1,68 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>([]);
+  const [filterTodos, setFilterTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<string>('all');
+  const [searchFilter, setSearchFilter] = useState<string>('');
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const [loadingModal, setLoadingModal] = useState(true);
+
+  const [modal, setModal] = useState(false);
+
+  useEffect(() => {
+    if (selectedTodo?.userId) {
+      setLoadingModal(true);
+      getUser(selectedTodo?.userId).then(fetchedUser => setUser(fetchedUser));
```

Comment:
The email address in the anchor tag should be dynamically set based on the `user` object to ensure it reflects the correct user's email. Please update this logic accordingly.